src/handler/http/request/prom/mod.rs

```diff
@@ -587,27 +586,15 @@ fn validate_metadata_params(
             }
         },
     };
-    let start = match start {
-        None => 0,
-        Some(s) => match parse_str_to_timestamp_micros(&s) {
-            Ok(t) => t,
-            Err(e) => {
-                let err = format!("parse time error: {e}");
-                log::error!("{err}");
-                return Err(err);
-            }
-        },
+    let start = if start.is_none() || start.as_ref().unwrap().is_empty() {
+        0
+    } else {
+        parse_str_to_timestamp_micros(&start.unwrap()).map_err(|e| e.to_string())?
     };
-    let end = match end {
-        None => i64::MAX,
-        Some(s) => match parse_str_to_timestamp_micros(&s) {
-            Ok(t) => t,
-            Err(e) => {
-                let err = format!("parse time error: {e}");
-                log::error!("{err}");
-                return Err(err);
-            }
-        },
+    let end = if end.is_none() || end.as_ref().unwrap().is_empty() {
```

Comment:
I think any input value of `end` > now() should be set to now().

It's also worth ensuring that start <= end. Do you agree?

Reply:
agree, we can fix it later. it can let our app stronger.